src/Umbraco.Web.BackOffice/Trees/ContentTreeControllerBase.cs

```diff
@@ -429,6 +429,24 @@ protected sealed override ActionResult<TreeNodeCollection> GetTreeNodes(string i
             return nodes ?? new TreeNodeCollection();
         }
 
+        // if we're rendering the root and people don't have access to the root still grant them access to the recycle bin
+        else if (id == Constants.System.RootString && !UserStartNodes.Contains(Constants.System.Root))
+        {
+            var nodes = new TreeNodeCollection();
+
+            nodes?.Add(CreateTreeNode(
```

Comment:
I would remove the "?" as we know that `nodes` cannot be null at this point.

Reply:
Agreed.  I've rewritten the code so this no longer applies.